### PR TITLE
Added/updated api test stubs for discovery feature

### PR DESCRIPTION
**Pull request by sghai**
Added/updated pxe-less tests for api's.

### CODE REVIEW

**Comment by oshtaier** [comment]
ACK

File: tests/foreman/api/test_discoveredhost.py [hunk 1]
```diff
@@ -293,23 +387,24 @@ def test_positive_create_discovery_rule_with_host_limit(self):
...
 
         @Status: Manual
         """
 
     @stubbed()
     @tier3
-    def test_positive_update_discovery_rule_provision(self):
-        """Update the discovered host name and provision it
+    def test_positive_provision_with_updated_hostname_in_rule(self):
+        """Update the discovered hostname in exiting rule and provision a host
```
Comment (sthirugn) [inline tests/foreman/api/test_discoveredhost.py]: s/exiting/existing

Comment (sghai) [inline tests/foreman/api/test_discoveredhost.py]: :+1:

**Comment by sthirugn** [comment]
ACK

**Pull request merged**